snuba/query/snql/parser.py

```diff
@@ -988,6 +988,40 @@ def transform(exp: Expression) -> Expression:
     query.transform_expressions(transform)
 
 
+def _treeify_or_and_conditions(
+    query: Union[CompositeQuery[QueryEntity], LogicalQuery]
+) -> None:
+    """
+    look for expressions like or(a, b, c) and turn them into or(a, or(b, c))
+                              and(a, b, c) and turn them into and(a, and(b, c))
+
+    even though clickhouse sql supports arbitrary amount of arguments there are other parts of the
+    codebase which assume `or` and `and` have two arguments
+
+    Adding this post-process step is easier than changing the rest of the query pipeline
+    """
+
+    def transform(exp: Expression) -> Expression:
+        if not isinstance(exp, FunctionCall):
+            return exp
+
+        if exp.function_name not in ("and", "or"):
+            return exp
+        args = exp.parameters
```

Comment:
There are helper functions that will do this for you already: https://github.com/getsentry/snuba/blob/master/snuba/query/conditions.py#L278